src/commands/backlog/checkout.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * https://docs.atlassian.com/bitbucket-server/rest/7.6.0/bitbucket-branch-rest.html
+ * https://bwa.nrs.gov.bc.ca/int/jira/rest/api/2/issue/ONETEAM-275
+ * https://bwa.nrs.gov.bc.ca/int/jira/rest/dev-status/latest/issue/summary?issueId=131578
+ * https://bwa.nrs.gov.bc.ca/int/jira/rest/dev-status/1.0/issue/detail?issueId=131578&applicationType=stash&dataType=pullrequest
+ */
+import {flags} from '@oclif/command'
+import {AxiosJiraClient} from '../../api/service/axios-jira-client'
+import {GitBaseCommand} from '../../git-base'
+import * as inquirer from 'inquirer'
+import * as path from 'path'
+import {BranchReference, RepositoryReference} from '../../api/service/axios-bitbucket-client'
+import cli from 'cli-ux'
+
+export default class GitCheckout extends GitBaseCommand {
+  static description = 'Create (if required), and checkout the git branch supporting a Jira issue (bug, new feature, improvement, etc...)'
+
+  static flags = {
+    project: flags.string({hidden: true, char: 'p', description: 'BitBucket Project/Group Name'}),
+    repository: flags.string({hidden: true, char: 'r', description: 'BitBucket Repository Name'}),
+    branch: flags.string({hidden: true, char: 'b', description: 'Remote Branch Name'}),
+    username: flags.string({hidden: true, char: 'u', description: 'Username to be appended to branch names'}),
+  }
+
+  static args = [{name: 'issue', description: 'Jira issue key (e.g.: WEBADE-123)'}]
+
+  async action(message: string, promise: Promise<any>) {
+    return Promise.resolve(true)
+    .then(() => {
+      cli.action.start(message)
+      return promise
+    }).finally(() => {
+      cli.action.stop()
+    })
+  }
+
+  async run() {
```

Comment:
Same as checkin.ts . This is not very readable and at a glance I can just say it does stuff. No idea if it doing the right thing.